src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs

```diff
@@ -32,41 +32,65 @@ public override HttpRequestMessage CreateRequest(ServerNode node, out string url
 
             var request = new HttpRequestMessage
             {
-                Method = HttpMethod.Post
-            };
-
-            var commands = new List<DynamicJsonValue>();
-            foreach (var command in _commands)
-            {
-                var cacheKey = GetCacheKey(command, out string _);
-                using (_cache.Get(_context, cacheKey, out long cachedEtag, out BlittableJsonReaderObject _))
+                Method = HttpMethod.Post,
+                Content = new BlittableJsonContent(stream =>
                 {
-                    var headers = new DynamicJsonValue();
-                    if (cachedEtag != 0)
-                        headers["If-None-Match"] = $"\"{cachedEtag}\"";
-
-                    foreach (var header in command.Headers)
-                        headers[header.Key] = header.Value;
-                    commands.Add(new DynamicJsonValue
+                    using (var writer = new BlittableJsonTextWriter(_context, stream))
                     {
-                        [nameof(GetRequest.Url)] = $"/databases/{node.Database}{command.Url}",
-                        [nameof(GetRequest.Query)] = $"{command.Query}",
-                        [nameof(GetRequest.Method)] = command.Method,
-                        [nameof(GetRequest.Headers)] = headers,
-                        [nameof(GetRequest.Content)] = command.Content
-                    });
-                }
-            }
-
-            request.Content = new BlittableJsonContent(stream =>
-            {
-                using (var writer = new BlittableJsonTextWriter(_context, stream))
-                {
-                    writer.WriteStartObject();
-                    writer.WriteArray("Requests", commands, _context);
-                    writer.WriteEndObject();
-                }
-            });
+                        writer.WriteStartObject();
+                        writer.WriteArray(_context, "Requests", _commands, (w, c, command) =>
```

Comment:
We probably want to avoid this kind of code. It requires us to do a delegate invocation per command.
If we turn that into a simpler `foreach` it would do less.